leif/parse_slo_rules.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package leif
+
+import (
+	"encoding/json"
+	"errors"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var dayReg = regexp.MustCompile(`[0-9]+d`)
```

Comment:
This regex is still a little permissive. It will match  `1d1h2m`, which will, when used by the ReplaceAllStringFunc will return `24h1h2m`, which is a parse error. I'd recommend adding anchors to the start and end to make this a more strict regex:

```golang
regexp.MustCompile(`^\s*[0-9]+d\s*$`)
```

This represents "start of string" **^** followed by any ammount of whitespace **\s*** , followed by your original regex, then ending with any ammount of whitespace **\s*** and end of line **$**

Reply:
Oh, I purposefully allowed values like `24h1h2m` since they are allowed by the time library. With those additions, values like `1s1d` would also not pass while the time library allows `1s1h`. I'm fine with changing this to make users strictly start with days, followed by hours and so on, and then when parsing add these to any hour values (so `1d1h2m` > `25h2m` and `1s1h` > error). What do you think?

Reply:
Oh snap; I must have gotten this confused with another language. 

The failure I was expecting was that if you specify the same unit of time multiple times, then it is an error e.g. `1h1h1h` => error as opposed to just `3h`. I think this will be fine for us to leave in for now, but I suspect this wont be used too frequently as it will (probably) break the typescript bot.